test/extended/util/disruption/backend_sampler_tester.go

```diff
@@ -0,0 +1,130 @@
+package disruption
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/onsi/ginkgo"
+	"github.com/openshift/origin/pkg/monitor"
+	"k8s.io/client-go/kubernetes/scheme"
+	"k8s.io/client-go/tools/events"
+	"k8s.io/kubernetes/test/e2e/framework"
+	"k8s.io/kubernetes/test/e2e/upgrades"
+)
+
+type BackendSampler interface {
+	GetDisruptionBackendName() string
+	GetLocator() string
+	GetURL() (string, error)
+	StartEndpointMonitoring(ctx context.Context, m monitor.Recorder, eventRecorder events.EventRecorder) error
+}
```

Comment:
Could this interface move over to the package that implements it, it's a little off in the weeds here.

Reply:
> Could this interface move over to the package that implements it, it's a little off in the weeds here.

it's a cyclical import thing and golang encourages local interfaces with duck typing.

Reply:
Ah local to where it's used, makes sense.